apps/cnspec/cmd/policy.go

```diff
@@ -0,0 +1,747 @@
+	"go.mondoo.com/cnspec/v9/internal/bundle"
+	"go.mondoo.com/cnspec/v9/policy"
+	cnspec_upstream "go.mondoo.com/cnspec/v9/upstream"
+	"gopkg.in/yaml.v3"
+	"k8s.io/utils/ptr"
+)
+
+const (
+	PolicyMrnPrefix          = "//policy.api.mondoo.app"
+	PrivatePoliciesMrnPrefix = "//policy.api.mondoo.app/spaces"
+)
+
+func init() {
+	rootCmd.AddCommand(policyCmd)
+	policyCmd.AddCommand(policyDeleteCmd)
+	policyCmd.AddCommand(policyEnableCmd)
+	policyCmd.AddCommand(policyDisableCmd)
+	policyCmd.AddCommand(policyInitCmd)
+
+	// list
+	policyCmd.AddCommand(policyListCmd)
+	policyListCmd.Flags().StringP("file", "f", "", "a local bundle file")
+
+	// upload
+	policyUploadCmd.Flags().Bool("no-lint", false, "Disable linting of the bundle before publishing.")
+	policyUploadCmd.Flags().String("policy-version", "", "Override the version of each policy in the bundle.")
+	policyCmd.AddCommand(policyUploadCmd)
+
+	// lint
+	policyLintCmd.Flags().StringP("output", "o", "cli", "Set output format: compact, sarif")
+	policyLintCmd.Flags().String("output-file", "", "Set output file")
+	policyCmd.AddCommand(policyLintCmd)
+
+	// fmt
+	policyFmtCmd.Flags().Bool("sort", false, "sort the bundle.")
+	policyCmd.AddCommand(policyFmtCmd)
+
+	// info
+	policyInfoCmd.Flags().StringP("file", "f", "", "a local bundle file")
+	policyCmd.AddCommand(policyInfoCmd)
+
+	// download
+	policyDownloadCmd.Flags().StringP("file", "f", "", "output file")
+	policyCmd.AddCommand(policyDownloadCmd)
+
+	// docs
+	policyDocsCmd.Flags().Bool("no-code", false, "enable/disable code blocks inside of docs")
+	policyDocsCmd.Flags().Bool("no-ids", false, "enable/disable the printing of ID fields")
+	policyCmd.AddCommand(policyDocsCmd)
+}
+
+var policyCmd = &cobra.Command{
+	Use:   "policy",
+	Short: "Manage local and upstream policies.",
+}
+
+var policyListCmd = &cobra.Command{
+	Use:   "list [-f bundle]",
+	Short: "List enabled policies in the connected space.",
+	Args:  cobra.MaximumNArgs(0),
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		if err := viper.BindPFlag("file", cmd.Flags().Lookup("file")); err != nil {
+			return err
+		}
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		bundleFile := viper.GetString("file")
+		var policies []*policy.Policy
+		if bundleFile != "" {
+			policyBundle, err := policy.DefaultBundleLoader().BundleFromPaths(bundleFile)
+			if err != nil {
+				return err
+			}
+			policies = policyBundle.Policies
+		} else {
+			opts, err := config.Read()
+			if err != nil {
+				return err
+			}
+			config.DisplayUsedConfig()
+
+			mondooClient, err := getGqlClient(opts)
+			if err != nil {
+				return err
+			}
+
+			policies, err = cnspec_upstream.SearchPolicy(
+				context.Background(), mondooClient, opts.GetParentMrn(), ptr.To(true), ptr.To(true), ptr.To(true))
+			if err != nil {
+				return err
+			}
+		}
+		for _, policy := range policies {
+			fmt.Println(policy.Name + " " + policy.Version)
```

Comment:
```
My Policy Title v1.0.0 (enabled, private)
  //policy.api.mondoo.app/policies/mondoo-kubernetes-security
  
My Policy Title v1.0.0 (preview)
  //policy.api.mondoo.app/policies/mondoo-kubernetes-security
```

There are three different scenarios that we want to handle here:

1. Policy is enabled or in preview, then we show it in brackets
2. If a policy is not enabled we do not show anything
3. If we policy is private, we display private, otherwise we do not add text

Reply:
Do we have preview policies?